src/elements/common/sub-header/GridViewButton.js

```diff
@@ -0,0 +1,29 @@
+// @flow
+
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import Button from '../../../components/button';
+import IconGridViewInverted from '../../../icons/general/IconGridViewInverted';
+import IconGridView from '../../../icons/general/IconGridView';
+import messages from '../messages';
+import Tooltip from '../Tooltip';
+import './GridViewButton.scss';
+
+type Props = {
+    isGridView: boolean,
+};
+
+const GridViewButton = (props: Props) => {
+    const { isGridView, ...rest } = props;
+    const icon = isGridView ? <IconGridView width={17} height={17} /> : <IconGridViewInverted width={17} height={17} />;
+
+    return (
+        <Tooltip text={<FormattedMessage {...messages.gridView} />}>
+            <Button className="be-btn-grid-view" aria-label={messages.gridView.defaultMessage} type="button" {...rest}>
```

Comment:
We should be passing each message to `react-intl` so it can be rendered in the user's selected locale. Referencing the `defaultMessage` directly willy likely only show the English copy.

Reply:
Do we need a corresponding message for switching back to list view once grid view is toggled?